src/puppetlabs/puppetdb/command.clj

```diff
@@ -235,16 +235,16 @@
 (defn store-report*
   [version db {:keys [payload annotations]}]
   (let [id          (:id annotations)
-        report      (upon-error-throw-fatality
-                     (report/validate! version payload))
-        certname    (:certname report)
+        {:keys [certname puppet_version] :as report} (->> payload
+                                                          (s/validate report/report-schema)
```

Comment:
Is there a better way to avoid an explicit call to s/validate? For example, we could switch store-report* or its upstream to be a defn-validated.